telemetry/appblock.go

```diff
@@ -0,0 +1,105 @@
+package telemetry
+
+import (
+	"runtime"
+	"strconv"
+	"time"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/mempool"
+	"github.com/hashicorp/go-metrics"
+)
+
+type BlockTelemetry struct {
+	startTime            time.Time
+	initialBlockMemAlloc uint64
+	blockTxsSize         int
+}
+
+// InitBlockMetrics initializes the block metrics for the application
+// by initializing a BlockTelemetry struct and setting the prepare proposal,
+// begin blocker, and end blocker handlers with the metrics.
+func InitBlockMetrics(app *baseapp.BaseApp) {
```

Comment:
Init `InitAppMetrics` instead? Feel slike we'll be adding more things here and the block is just a part of it.